app/domain/authentication/authn_oidc/v2/strategy.rb

```diff
@@ -2,52 +2,89 @@
 module Authentication
   module AuthnOidc
     module V2
-      class Strategy
-        def initialize(
-          authenticator:,
-          client: Authentication::AuthnOidc::V2::Client,
-          logger: Rails.logger
-        )
-          @authenticator = authenticator
-          @client = client.new(authenticator: authenticator)
-          @logger = logger
-        end
+      module Strategies
+        module Common
```

Comment:
Please NEVER use modules to inject methods into downstream objects, particularly the initializer.  Modules make it nearly impossible to reason about where a classes methods come from.  

Instead either use inheritance (probably the correct solution for this section), or create a generic, stateless object "Utility" object with common methods.  Utility objects should follow adhere to functional programming best practices (stateless, no mutation, etc.).